cpp/marc_tools/marc_augmentor.cc

```diff
@@ -373,6 +386,39 @@ bool ReplaceField(MARC::Record * const record, const MARC::Tag &tag, const char
 }
 
 
+bool ReplaceFieldRegex(MARC::Record * const record, const MARC::Tag &tag, const char subfield_code, const std::string &replacement_regex,
+                       CompiledPattern * const condition = nullptr) {
+    if (condition != nullptr) {
+        if (not condition->matched(*record))
+            return false;
+    }
+
+    bool replaced_at_least_one(false);
+    for (auto &field : *record) {
+        if (field.getTag() != tag)
+            continue;
+
+        MARC::Subfields subfields(field.getContents());
+        if (subfields.hasSubfield(subfield_code))
+            LOG_ERROR("Regex Replacement with subfield not supported! Please use '--replace-subfield-if-regex' instead.");
+
+        std::string field_value(field.getContents());
+        bool global(HasGlobalFlag(replacement_regex));
+        std::vector<std::string> pattern_and_replacement(GetSplitReplacementRegexWithoutGlobalFlagAndEscapedBackrefs(replacement_regex));
+
+        if (unlikely(pattern_and_replacement.size() != 2))
+            LOG_ERROR("Invalid replacement pattern :\"" + replacement_regex + "\"\nMust follow /pattern/replacement/g? scheme");
+
+        ThreadSafeRegexMatcher replace_matcher(pattern_and_replacement[0]);
+        const std::string replacement_text(replace_matcher.replaceWithBackreferences(field_value, pattern_and_replacement[1], global));
+
+        field.setContents(replacement_text);
+        replaced_at_least_one = true;
+    }
+
+    return replaced_at_least_one;
+}
+
```

Comment:
This function is very similar to ReplaceSubfieldRegex(). Maybe it is possible to factor out some aspects or to unify and reduce the addition of subfields to the addition of fields.